python/modules/map_plate.py

```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
-    names_path = FM.path_join(mp_path, "args_ind.csv")
-    positions_path = FM.path_join(mp_path, "args_names.csv")
-    if FM.path_check_existence(positions_path):
-        tmp = True
-    active_wells = _parsing_matrix(active_path, deltimer, exp_part)
-    for well in active_wells:
-        name = _collecting_exp_settings(names_path, well, deltimer)
-        result = []
-        if tmp == True:
-            result.append(_collecting_exp_settings(positions_path, well, deltimer))
-        else:
-            result.append(name)
-        result.append(exp_name)
-        result.append(exp_parts_all)
-        result.append(exp_part)
-        for mp_file in paths_mp:
-            param = _collecting_exp_settings(mp_file, well, deltimer)
-            result.append(param)
-        param_dict[name] = deltimer.join(result)
-    return param_dict
-  
-def _collecting_exp_settings(mp_file, well, deltimer):
-    row_nr, col_nr = well
-    data = CSV_M.read_csv(mp_file, deltimer)
-    print(mp_file)
-    print(row_nr)
-    print(col_nr)
-    return (data[row_nr][col_nr])
-    
-def _getting_column_titles(abs_path, paths):
+def get_param_value(mp_dict, well_name, param): # get values for a given well and parameter
+    value = mp_dict[well_name][param]
+    return value
+
+def get_param_all_values(mp_dict, param): # get all values for a given param from map_plate dictionary
+    values = []
+    for well in mp_dict:
+        values.append(mp_dict[well][param])
+    return values
+
+def get_param_unique_values(mp_dict, param): # get all unique values for a given param from map_plate dictionary
+    values = get_param_all_values(mp_dict, param)
+    unique = list(set(values))
+    return unique
+
+def get_all_params_names(mp_dict): # get all key names from map_plate dictionary
     names = []
-    names.append("position.name")
-    names.append("exp.id")
-    names.append("exp.parts")
-    names.append("exp.part.id")
-    for path in paths:
-        rel_path = FM.path_get_relative(abs_path, path)
-        fullname = ".".join(FM._path_split(rel_path))
-        extension_length = len(FM.file_get_extension(fullname))  #extracting proper column name from rel_path i.e. compare.1.1 from compare/1.1.csv
-        name = fullname[:-extension_length]
-        names.append(name)
-    #names = ",".join(names)
+    key_0 = list(mp_dict.keys())[0]
+    names = list(mp_dict[key_0].keys()) # if we assume that number of information between wells is constant
+    """params_count = len(mp_dict[key_0])
+    for well in mp_dict: # if we assume that number of information about wells may somehow differ
+        tmp= len(mp_dict[well])
+        if params_count != tmp:
+            logger.warning("Some wells are not described by the same number of information: %s, %s", key_0, well)
+            if tmp > params_count:
+                names = []
+                params_count = tmp
+                for param in mp_dict[well]:
+                    names.append(param)"""
```

Comment:
@stork119 I left if because map_plate is still in development stage and I wasn't sure which assumption will meet all possible cases. If we decide that number of information about wells may somehow differ, we can use that (already written) code.